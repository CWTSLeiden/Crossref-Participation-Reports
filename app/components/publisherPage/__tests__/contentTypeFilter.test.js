import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon'
import ContentTypeFilter from "../components/checksFilter"






//Dummy Data
const filters = ['journal-articles', 'books', 'book-chapters', 'conference-papers', 'reports']
let currentFilter = 'journal-articles'

//Mocks
const sandbox = sinon.sandbox.create()
const setState = sandbox.spy(ContentTypeFilter.prototype, 'setState')
const setFilter = sandbox.stub()




//Render and begin tests
const shallowComponent = shallow(<ContentTypeFilter filters={filters} currentFilter={currentFilter} setFilter={setFilter} />)




test('shallow snapshot', () => {
  expect(toJson(shallowComponent)).toMatchSnapshot()
})


describe('open filter menu', () => {
  test('menu should render', () => {
    shallowComponent.find('.contentTypeFilter').prop('onClick')()
    expect(setState.callCount).toBe(1)
    shallowComponent.update()
    expect(shallowComponent.state('menuOpen')).toBe(true)
    const filterList = shallowComponent.find('.filterList')
    expect(filterList.children().length).toBe(filters.length)
    expect(toJson(shallowComponent)).toMatchSnapshot()
  })
  
  test('current filter should be checked', () => {
    expect(shallowComponent.find(`#${currentFilter}_checkmark`).length).toBe(1)
  })
})


describe('select a filter', () => {
  let newFilter = filters[1]

  test('filterMenu should close and setFilter called with new filter', () => {
    sandbox.resetHistory()
    shallowComponent.find(`#${newFilter}_button`).prop('onClick')()
    shallowComponent.find('.contentTypeFilter').prop('onClick')()
    expect(setState.callCount).toBe(1)
    expect(setFilter.callCount).toBe(1)
    expect(setFilter.calledWith(newFilter))
    shallowComponent.setProps({currentFilter: newFilter})
    shallowComponent.update()
    expect(shallowComponent.state('menuOpen')).toBe(false)
    const filterList = shallowComponent.find('.filterList')
    expect(filterList.children().length).toBe(0)
  })


  test('new filter has checkmark when menu opened', () => {
    shallowComponent.find('.contentTypeFilter').prop('onClick')()
    shallowComponent.update()
    expect(shallowComponent.find(`#${newFilter}_checkmark`).length).toBe(1)
  })
})