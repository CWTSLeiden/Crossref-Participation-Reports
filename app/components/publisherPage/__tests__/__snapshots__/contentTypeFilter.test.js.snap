// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open filter menu menu should render 1`] = `
<div
  className="contentTypeFilter"
  onBlur={[Function]}
  onClick={[Function]}
  tabIndex="0"
>
  Content Type
  <img
    src="/participationreports//images/Asset_Icons_Grey_Chevron.png"
    style={
      Object {
        "height": "20px",
        "marginTop": "-2px",
        "width": "20px",
      }
    }
  />
  <div
    className="filterList"
  >
    <div
      className="filterButton"
      id="journal-articles_button"
      key="journal-articles"
      onClick={[Function]}
    >
      <div
        className="checkmark"
      >
        <img
          id="journal-articles_checkmark"
          src="/participationreports//images/iconmonstr-check-mark-1.svg"
          style={
            Object {
              "height": "14px",
              "width": "14px",
            }
          }
        />
      </div>
      <div
        className="buttonText"
      >
        Journal articles
      </div>
    </div>
    <div
      className="filterButton"
      id="books_button"
      key="books"
      onClick={[Function]}
    >
      <div
        className="checkmark"
      />
      <div
        className="buttonText"
      >
        Books
      </div>
    </div>
    <div
      className="filterButton"
      id="book-chapters_button"
      key="book-chapters"
      onClick={[Function]}
    >
      <div
        className="checkmark"
      />
      <div
        className="buttonText"
      >
        Book chapters
      </div>
    </div>
    <div
      className="filterButton"
      id="conference-papers_button"
      key="conference-papers"
      onClick={[Function]}
    >
      <div
        className="checkmark"
      />
      <div
        className="buttonText"
      >
        Conference papers
      </div>
    </div>
    <div
      className="filterButton"
      id="reports_button"
      key="reports"
      onClick={[Function]}
    >
      <div
        className="checkmark"
      />
      <div
        className="buttonText"
      >
        Reports
      </div>
    </div>
  </div>
</div>
`;

exports[`shallow snapshot 1`] = `
<div
  className="contentTypeFilter"
  onBlur={[Function]}
  onClick={[Function]}
  tabIndex="0"
>
  Content Type
  <img
    src="/participationreports//images/Asset_Icons_Grey_Chevron.png"
    style={
      Object {
        "height": "20px",
        "marginTop": "-2px",
        "width": "20px",
      }
    }
  />
</div>
`;
